### PR TITLE
EZP-31482: Introduced strict types for FieldType & VO

### PR DESCRIPTION
**Pull request by adamwojs**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-31482](https://jira.ez.no/browse/EZP-31482)
| **Bug/Improvement**| yes
| **New feature**    | no
| **Target version** | `master`
| **BC breaks**      | yes
| **Tests pass**     | yes
| **Doc needed**     | yes

This PR introduces strict types for the `\eZ\Publish\API\Repository\FieldType` and `eZ\Publish\API\Repository\Values\**`. The most important changes are:

* Added `declare(strict_types=1)` and and type hints for methods parameters and return value
* Aligned licence header
* Added visibility modifier to `const`
* Changed `$valueFormat` type from `string` to `int` in `\eZ\Publish\API\Repository\Values\Content\Query\Criterion\Operator\Specifications::__construct`  
* `\eZ\Publish\API\Repository\Values\User\Policy::getLimitations` will no longer return '*'

**TODO**:
- [x] Introduce strict types for `\eZ\Publish\API\Repository\Values\Content\Query\**`
- [x] Implement feature / fix a bug.
- [x] Implement tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: eZ/Publish/API/Repository/Values/Content/Content.php [hunk 1]
```diff
@@ -56,18 +56,18 @@ public function getName($languageCode = null)
      * On non translatable fields this method ignores the languageCode parameter, and return main language field value.
      *
      * @param string $fieldDefIdentifier
-     * @param string $languageCode
+     * @param string|null $languageCode
      *
      * @return \eZ\Publish\SPI\FieldType\Value|null a primitive type or a field type Value object depending on the field type.
      */
-    abstract public function getFieldValue($fieldDefIdentifier, $languageCode = null);
+    abstract public function getFieldValue(string $fieldDefIdentifier, ?string $languageCode = null): ?\eZ\Publish\SPI\FieldType\Value;
```
Comment (mikadamczyk) [inline eZ/Publish/API/Repository/Values/Content/Content.php]:
```suggestion
    abstract public function getFieldValue(string $fieldDefIdentifier, ?string $languageCode = null): ?Value;
```

File: eZ/Publish/API/Repository/Values/Content/Query/Criterion/LogicalOperator.php [hunk 2]
```diff
@@ -55,10 +55,11 @@ public function __construct(array $criteria)
     }
 
     /**
+     * @return array
```
Comment (mikadamczyk) [inline eZ/Publish/API/Repository/Values/Content/Query/Criterion/LogicalOperator.php]:
```suggestion
```

File: eZ/Publish/API/Repository/Values/Content/Query/Criterion/Operator/Specifications.php [hunk 3]
```diff
@@ -67,11 +67,11 @@ class Specifications
      * Creates a new Specifications object.
      *
      * @param string $operator The specified operator, as one of the Operator::* constants
-     * @param string $valueFormat The accepted value format, either {@see self::FORMAT_ARRAY} or {@see self::FORMAT_SINGLE}
+     * @param int $valueFormat The accepted value format, either {@see self::FORMAT_ARRAY} or {@see self::FORMAT_SINGLE}
      * @param int $valueTypes The supported value types, as a bit field of the {@see self::TYPE_*} constants
```
Comment (mikadamczyk) [inline eZ/Publish/API/Repository/Values/Content/Query/Criterion/Operator/Specifications.php]:
```suggestion
     * @param int|null $valueTypes The supported value types, as a bit field of the {@see self::TYPE_*} constants
```

File: eZ/Publish/Core/Repository/Values/User/UserGroup.php [hunk 4]
```diff
@@ -31,7 +33,7 @@ class UserGroup extends APIUserGroup
      *
      * @return \eZ\Publish\API\Repository\Values\Content\VersionInfo
      */
-    public function getVersionInfo()
+    public function getVersionInfo(): \eZ\Publish\API\Repository\Values\Content\VersionInfo
```
Comment (mikadamczyk) [inline eZ/Publish/Core/Repository/Values/User/UserGroup.php]:
```suggestion
    public function getVersionInfo(): VersionInfo
```

File: eZ/Publish/Core/Repository/Permission/PermissionCriterionResolver.php [hunk 5]
```diff
@@ -78,7 +78,7 @@ public function getPermissionsCriterion($module = 'content', $function = 'read',
             /** @var \eZ\Publish\API\Repository\Values\User\Policy */
             foreach ($permissionSet['policies'] as $policy) {
                 $limitations = $policy->getLimitations();
-                if ($limitations === '*' || empty($limitations)) {
+                if (empty($limitations)) {
```
Comment (alongosz) [inline eZ/Publish/Core/Repository/Permission/PermissionCriterionResolver.php]: :sparkling_heart:

**Pull request merged**